### PR TITLE
Feature: distinguish commitment venues in the request form

### PR DESCRIPTION
**Pull request by celestemartinez**
This PR makes the following changes to the submission invitation if the venue is a commitment venue:
- set signatures to Super User
- adds paper_link to content
- add preprocess

### CODE REVIEW

File: openreview/conference/helpers.py [hunk 1]
```diff
@@ -434,6 +434,8 @@ def get_submission_stage(request_forum, venue):
     else:
         withdraw_submission_exp_date = None
 
+    commitments_venue = request_forum.content.get('commitments_venue', 'No') == 'Yes'
```
Comment (melisabok) [inline openreview/conference/helpers.py:437]: can we call it `ARR_commitments_venue`?

File: openreview/stages/venue_stages.py [hunk 2]
```diff
@@ -293,6 +295,20 @@ def get_content(self, api_version='1', conference=None, venue_id=None):
                     }
                 }
 
+            if self.commitments_venue:
```
Comment (melisabok) [inline openreview/stages/venue_stages.py:298]: I would check if paper_link is already in the content and don't overwrite it, the PCs might want to change the description.

File: openreview/venue/process/submission_commitments_preprocess.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    paper_link = edit.note.content['paper_link']['value']
+    paper_forum = paper_link.split('=')[-1]
+
+    try:
+        client_v1=openreview.Client(baseurl=openreview.tools.get_base_urls(client)[0], token=client.token)
+        arr_submission = client_v1.get_note(paper_forum)
+    except openreview.OpenReviewException as e:
+        raise openreview.OpenReviewException('Provided paper link does not correspond to a submission in OpenReview')
+
+    valid_prefixes = ['aclweb.org/ACL/ARR/2021', 'aclweb.org/ACL/ARR/2022', 'aclweb.org/ACL/ARR/2023', 'aclweb.org/ACL/ARR/2024']
```
Comment (melisabok) [inline openreview/venue/process/submission_commitments_preprocess.py:12]: what if we have one single prefix `aclweb.org/ACL/ARR`?

File: openreview/conference/helpers.py (hunk 1 above)
Comment (melisabok) [reply openreview/conference/helpers.py:437]: is it possible that we would receive commitments from other venues?

**Comment by melisabok** [comment]
The changes look good! thanks Celeste for making this PR so quickly. 

I would suggest to put the migration code here so we can have an invitation that runs after the submission deadline and migrates the papers and reviews. What do you think?

File: openreview/venue/process/submission_commitments_preprocess.py (hunk 3 above)
Comment (celestemartinez) [reply openreview/venue/process/submission_commitments_preprocess.py:12]: Yes! Not sure why we were using the year here...I don't see any other venue ids that would start with this.

File: openreview/conference/helpers.py (hunk 1 above)
Comment (racheljsmart) [reply openreview/conference/helpers.py:437]: possibly from EACL (which themselves are ARR commitments)

File: openreview/stages/venue_stages.py [hunk 4]
```diff
@@ -293,6 +295,20 @@ def get_content(self, api_version='1', conference=None, venue_id=None):
...
+                content['paper_link'] = {
+                    'value': {
+                        'param': {
+                            'type': 'string',
+                            'regex': '(http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+)',
+                            'mismatchError': 'must be a valid link to an OpenrReview submission: https://openreview.net/forum?id=...'
+                        }
+                    },
+                    'description': 'Please provide the link to your ARR submission.',
```
Comment (racheljsmart) [inline openreview/stages/venue_stages.py:307]: Can "The link should have the following format: https://openreview.net/forum?id=<PAPER_ID>" be added to the description?

Comment (racheljsmart) [reply openreview/stages/venue_stages.py:307]: Does <PAPER_ID> make sense?

**Comment by celestemartinez** [comment]
> The changes look good! thanks Celeste for making this PR so quickly.
> 
> I would suggest to put the migration code here so we can have an invitation that runs after the submission deadline and migrates the papers and reviews. What do you think?

Yes, I agree. I can add this.

File: openreview/conference/helpers.py (hunk 1 above)
Comment (celestemartinez) [reply openreview/conference/helpers.py:438]: should we change this to `ARR_commitments_venue`? It does seem like all commitments we are getting so far come from ARR.

**Review by melisabok — APPROVED** [review]

File: openreview/conference/helpers.py (hunk 1 above)
Comment (melisabok) [reply openreview/conference/helpers.py:438]: sorry, I saw this too late. We can keep it as it is for now.

**Pull request merged**